promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
+				return
+			}
+		}
+
+		out, err := os.Create(filename)
```

Comment:
Choose a different name for your output file handle. This is too easy to confuse with `*out` which is the flag for the custom metric filename.

Reply:
Change file name -- >"TarFileOut" instead of out